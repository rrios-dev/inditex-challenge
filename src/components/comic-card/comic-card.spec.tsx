import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import ComicCard, { ComicCardProps } from ".";

describe("ComicCard component", () => {
  it("renders loading skeleton when status is loading", () => {
    const { asFragment } = render(<ComicCard status="loading" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error message when status is error", () => {
    const { asFragment } = render(<ComicCard status="error" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders comic card with correct props when status is success", () => {
    const mockProps: ComicCardProps = {
      status: "success",
      releaseDate: "2022-01-01",
      imageSrc: "path/to/image.jpg",
      imageAlt: "Alt text",
      title: "Comic Title",
    };
    const { asFragment } = render(<ComicCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
