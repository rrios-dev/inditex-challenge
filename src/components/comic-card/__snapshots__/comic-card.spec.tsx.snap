// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComicCard component > renders comic card with correct props when status is success 1`] = `
<DocumentFragment>
  <div
    class="_comic-card_400ff7 _flex_6eb20e _dir-column_6eb20e"
  >
    <img
      alt="Poster image of Comic Comic Title"
      class="_comic-card-image_400ff7"
      loading="lazy"
      src="path/to/image.jpg"
    />
    <p
      class="_typography_142002 _variant-body1_142002 color-fg _comic-card-title_400ff7"
    >
      Comic Title
    </p>
    <p
      class="_typography_142002 _variant-body1_142002 color-fg _comic-card-year_400ff7"
    >
      2022
    </p>
  </div>
</DocumentFragment>
`;

exports[`ComicCard component > renders error message when status is error 1`] = `
<DocumentFragment>
  <p
    class="_typography_142002 _variant-body1_142002 color-fg"
  >
    Error loading
  </p>
</DocumentFragment>
`;

exports[`ComicCard component > renders loading skeleton when status is loading 1`] = `
<DocumentFragment>
  <div
    class="_comic-card-skeleton_400ff7"
  >
    <div
      class="_comic-card-skeleton-image_400ff7"
    />
    <div
      class="_comic-card-skeleton-text_400ff7"
    />
    <div
      class="_comic-card-skeleton-year_400ff7"
    />
  </div>
</DocumentFragment>
`;
